pkg/liqo-controller-manager/internal-network/internalfabric-controller/route.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internalfabriccontroller
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	networkingv1alpha1 "github.com/liqotech/liqo/apis/networking/v1alpha1"
+	"github.com/liqotech/liqo/pkg/fabric"
+)
+
+func (r *InternalFabricReconciler) ensureRouteConfiguration(ctx context.Context, internalFabric *networkingv1alpha1.InternalFabric) error {
+	if internalFabric.Spec.Interface.Node.Name == "" {
+		return fmt.Errorf("internal fabric %q has node interface name empty", client.ObjectKeyFromObject(internalFabric))
+	}
+
+	route := &networkingv1alpha1.RouteConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GenerateRouteConfigurationName(internalFabric),
+			Namespace: internalFabric.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, route, func() error {
+		// Forge metadata
+		if route.Labels == nil {
+			route.Labels = make(map[string]string)
+		}
+		route.SetLabels(labels.Merge(route.Labels, fabric.ForgeRouteTargetLabels()))
```

Comment:
You can modify the forge function to return labels.Set instead of map[string]string